lib/ffi/sdr_funcs.go

```diff
@@ -663,3 +670,47 @@ func (sb *SealCalls) TreeD(ctx context.Context, sector storiface.SectorRef, unse
 
 	return nil
 }
+
+func (sb *SealCalls) SyntheticProofs(ctx context.Context, task *harmonytask.TaskID, sector storiface.SectorRef, sealed cid.Cid, unsealed cid.Cid, randomness abi.SealRandomness, pieces []abi.PieceInfo) error {
+	fspaths, pathIDs, releaseSector, err := sb.sectors.AcquireSector(ctx, task, sector, storiface.FTCache|storiface.FTSealed, storiface.FTNone, storiface.PathSealing)
+	if err != nil {
+		return xerrors.Errorf("acquiring sector paths: %w", err)
+	}
+	defer releaseSector()
+
+	ssize, err := sector.ProofType.SectorSize()
+	if err != nil {
+		return err
+	}
+
+	err = ffi.GenerateSynthProofs(sector.ProofType, sealed, unsealed, fspaths.Cache, fspaths.Sealed, sector.ID.Number, sector.ID.Miner, randomness, pieces)
+	if err != nil {
+		return xerrors.Errorf("generating synthetic proof: %w", err)
+	}
+
+	if err = ffi.ClearCache(uint64(ssize), fspaths.Cache); err != nil {
+		// Note: non-fatal error.
+		log.Warn("failed to GenerateSynthProofs(): ", err)
+		log.Warnf("num:%d tkt:%v, sealedCID:%v, unsealedCID:%v", sector.ID.Number, randomness, sealed, unsealed)
```

Comment:
Honestly this should be fatal. Removing stuff should never fail quietly, this can lead to us pulling in garbage into long-term storage.

All Seal calls should cope with edge-case inputs but at the same time should do their best to leave the data in clean state.